front: memorize if occurrence list open in train list

Synar opened the pull request:
Close #11788

The state of pacedTrainItem is lost when the component is no longer rendered, and we only render part of the list for performance reasons. Whether the occurrence list of a paced train is opened or not is thus lost when scrolling to another part of the train list.

To fix it, keep the set of pacedTrainItem id with opened occurrence list in the timetable parent component containing the train list, which is always rendered.

I may add another commit to use a set for selectedTimetableItemIds too.

Edit : added a commit to fix occurence -> occurrence spelling mistakes everywhere

Synar committed:
front: memorize if occurence list open in train list

The state of pacedTrainItem is lost when the component is no
longer rendered, and we only render part of the list for performance
reasons. Thu whether the occurence list of a paced train is opened or not
is lost when scrolling to another part of the train list.

Thus keep the set of pacedTrainItem id with opened occurence list
in the timetable parent component containing the train list,
which is always rendered.

Signed-off-by: Alice K. <alice.khoudli@gmail.com>

SharglutDev commented on front/src/modules/trainschedule/components/Timetable/Timetable.tsx:
```diff
@@ -202,6 +217,8 @@ const Timetable = ({
                   selectPacedTrainToEdit={selectTimetableItemToEdit}
                   handleSelectPacedTrain={handleSelectTimetableItem}
+                  isOccurrencesListOpen={expandedTimetableItemIds.has(timetableItem.id)}
+                  handleOpenOccurencesList={handleExpandTimetableItem}
```
```suggestion
                  handleOpenOccurrencesList={handleExpandTimetableItem}
```

Synar replied:
Nice catch, thanks!

Synar replied:
Uh, turns out this is a common spelling mistake in our codebase, I've added a commit to fix it. ^^''

SharglutDev reviewed (approved):
Very nice ✅

Synar committed:
front: fix occurrence spelling

occurence -> occurrence
occured -> occurred

Signed-off-by: Alice K. <alice.khoudli@gmail.com>

leovalais commented:
There's a missing typo left in editoast, cf [`d56dc16` (#12299)](https://github.com/OpenRailAssociation/osrd/pull/12299/commits/d56dc160d253ab618cd759c090eef51f49a05c72)

Synar commented:
> There's a missing typo left in editoast, cf [`d56dc16` (#12299)](https://github.com/OpenRailAssociation/osrd/pull/12299/commits/d56dc160d253ab618cd759c090eef51f49a05c72)

Good catch! I've dropped the editoast part since it's already been fixed, but I've added "occured" -> "occurred" spelling corrections to the front too.
